lib/search.rb

```diff
@@ -1069,18 +1065,50 @@ def posts_query(limit, type_filter: nil, aggregate_search: false)
           "(#{rank} * #{category_priority_weights})"
         end
 
+      category_search_priority = <<~SQL
+      (
+        CASE categories.search_priority
+        WHEN #{Searchable::PRIORITIES[:very_high]}
+        THEN 3
+        WHEN #{Searchable::PRIORITIES[:very_low]}
+        THEN 1
+        ELSE 2
+        END
+      )
+      SQL
+
       if aggregate_search
-        posts = posts.select("#{data_ranking} rank", "topics.bumped_at topic_bumped_at")
-          .order("rank DESC", "topic_bumped_at DESC")
+        posts = posts
+          .select(
+            "#{category_search_priority} category_search_priority",
+            "#{data_ranking} rank",
+            "topics.bumped_at topic_bumped_at"
+          )
+          .order(
+            "category_search_priority DESC",
```

Comment:
I worry about unforseen impacts perf wise of this change, let's time it before and after on a big site?

Reply:
I measured it against a search query on bbs and didn't notice any significant slow down in the search query. Sometimes it is 1ms slower but sometimes it runs just as fast as the previously query. There is definitely additional work to be done in this query since we have an extra CASE statement and an extra ordering to do but I don't think it is significant enough.